frame/support/src/weights.rs

```diff
@@ -650,6 +650,15 @@ pub struct WeightToFeeCoefficient<Balance> {
 /// A list of coefficients that represent one polynomial.
 pub type WeightToFeeCoefficients<T> = SmallVec<[WeightToFeeCoefficient<T>; 4]>;
 
+/// A trait that describes the weight to fee calculation.
+pub trait WeightToFee {
+	/// The type that is returned as result from calculation.
+	type Balance: BaseArithmetic + From<u32> + Copy + Unsigned;
+
+	/// Calculates the fee from the passed `weight`.
+	fn calc(weight: &Weight) -> Self::Balance;
```

Comment:
```suggestion
	/// Calculates the fee from the passed `weight`.
	fn weight_to_fee(weight: &Weight) -> Self::Balance;
```

opinionated, but I think `pub trait SomeName { fn some_name() {} }` is an acceptable idiom. Sticking to `.calc` seems like restricting yourself based on the previous `WeightToFeePolynomial`